File: AngularLovelyjubbly.Data.Sql.Models/Record.cs
namespace AngularLovelyjubbly.Data.Sql.Models
{
    public partial class Record
    {
        public Record()
        {
        }

        public int RecordId { get; set; }
        //foreign key
        public int RecordCategoryId { get; set; }
        //navigation property
        public virtual RecordCategory RecordCategory { get; set; }
        public double RecordAmount { get; set; }
        //foreign key
        public int SeasonId { get; set; }
        //navigation property
        public virtual Season Season { get; set; }
        //foreign key
        public int TeamId { get; set; }
        //navigation property
        public virtual Team Team { get; set; }
        //foreign key
        public int CoachId { get; set; }
        //navigation property
        public virtual Coach Coach { get; set; }
        public byte Rank { get; set; }
        public string RecordImage { get; set; }
        public string Comments { get; set; }

        //foreign key
        public int? WeekId { get; set; }
        //navigation property
        public virtual Week Week { get; set; }
    }
}

File: AngularLovelyjubbly/ClientApp/src/app/shared/models/record.ts
import { ISeason } from './season';
import { ITeam } from './team';
import { ICoach } from './coach';
import { IRecordCategory } from './recordcategory';
import { IWeek } from './week';

export interface IRecord {
    recordId: number;
    recordCategory: IRecordCategory;
    team: ITeam;
    coach: ICoach;
    season: ISeason;
    rank: number;
    recordAmount: number;
    recordImage: string;
    week: IWeek;
    comments: string;
}

export class Record implements IRecord {

    constructor(public recordId: number = 0,
        public recordCategory: IRecordCategory = null,
        public team: ITeam = null,
        public coach: ICoach = null,
        public season: ISeason = null,
        public rank: number = null,
        public recordAmount: number = null,
        public recordImage: string,
        public week: IWeek = null,
        public comments: string) {
    }
}
